src/debug.c

```diff
@@ -1874,24 +1877,21 @@ static void writeStacktraces(int fd, int uplevel) {
         serverLogRawFromHandler(LL_WARNING, "writeStacktraces(): Failed to get the process's threads.");
     }
 
-    stacktrace_data stacktraces[len_tids];
-    stacktraces_mempool = stacktraces;
-    memset(stacktraces, 0, sizeof(stacktrace_data) * len_tids);
-
-    /* restart mempool iterator*/
-    atomicSet(g_thread_ids, 0);
+    char buff[PIPE_BUF];
+    /* Clear the stacktraces pipe */
+    while (read(stacktrace_pipe[0], &buff, sizeof(buff)) > 0) {}
 
     /* ThreadsManager_runOnThreads returns 0 if it is already running */
     if (!ThreadsManager_runOnThreads(tids, len_tids, collect_stacktrace_data)) return;
 
     size_t skipped = 0;
 
     /* initialize a buffer of size 1024, as recommended by the libc manual on low-level directory access */
-    char buff[1024];
     pid_t calling_tid = syscall(SYS_gettid);
     /* for backtrace_data in backtraces_data: */
     for (size_t i = 0; i < len_tids; i++) {
-        stacktrace_data curr_stacktrace_data = stacktraces[i];
+        stacktrace_data curr_stacktrace_data = {0};
+        if (read(stacktrace_pipe[0], &curr_stacktrace_data, sizeof(curr_stacktrace_data)) == -1) {/* Avoid warning. */};
         /*ThreadsManager_runOnThreads might fail to collect the thread's data */
         if (0 == curr_stacktrace_data.trace_size) {
```

Comment:
i think this needs an update.
if `read` failed, then we'll still have the last record in the buffer.

Reply:
already fixed :smile: